schema_salad/tests/test_cpp_codegen.py

```diff
@@ -22,28 +22,95 @@ def test_cwl_cpp_gen(tmp_path: Path) -> None:
 
 
 def test_cwl_cpp_generations(tmp_path: Path) -> None:
-    """End to end test of C++ generator using the CWL v1.0 schema."""
+    """End to end test of C++ generator using small scenarios."""
 
     test_dir = Path(cast(str, get_data("cpp_tests/01_single_record.yml"))).parents[0]
 
     # iterate through all cpp_tests YAML files
     for file in test_dir.iterdir():
-        if file.suffix != ".yaml":
+        if file.suffix != ".yml":
             continue
 
         # file with generated cpp output
         src_target = tmp_path / "test.h"
         # file with expected cpp output
-        expected = test_dir.with_suffix(".h")
+        expected = file.with_suffix(".h")
 
-        cpp_codegen(os.fspath(file), src_target)
+        cpp_codegen("file://" + os.fspath(file), src_target)
 
+        assert os.path.isfile(expected)
+        assert os.path.isfile(src_target)
         assert filecmp.cmp(expected, src_target, shallow=False)
 
 
+def test_cwl_cpp_generations_with_spdx(tmp_path: Path) -> None:
+    """End to end test of C++ generator checking for SPDX headers"""
+
+    src_target = tmp_path / "test.h"
+
+    input_file = cast(str, get_data("cpp_tests/01_single_record.yml"))
+
+    """Generating different combinations of license headers"""
+    """Generate License Identifier"""
+    cpp_codegen("file://" + input_file, src_target, spdx_license_identifier="Apache-2.0")
+    expected = input_file + "_with_license.h"
+    with open(expected, "r") as fin:
+        print(fin.read())
+
+    print("==============")
+    with open(src_target, "r") as fin:
+        print(fin.read())
+
+    assert os.path.exists(expected)
+    assert os.path.exists(src_target)
+    assert filecmp.cmp(expected, src_target, shallow=False)
```

Comment:
I'd rather you only check the first few lines for the expected content, as that is the focus of this test